.circleci/src/workflows/discovery.yml

```diff
@@ -3,53 +3,65 @@ jobs:
   - push-docker-image:
       name: push-discovery-provider
       context: [GCP, dockerhub]
-      instance-name: circleci-push-discovery-provider-$CIRCLE_BUILD_NUM
       service: discovery-provider
+      filters:
+        branches:
+          only: main
```

Comment:
i'm wondering why you added this here and in the creator and identity workflows? we want these to run on release branches as well. they're also sometimes used on feature branches just to have a tag pushed so someone can manually set it on a staging node for testing.

code pointer: see [here](https://github.com/AudiusProject/audius-protocol/blob/main/.circleci/config.yml#L425) for how it gets told to run it this workflow on release branches

Reply:
We can make it regex to include release branches. The rationale for filtering runs is that it feels wasteful because it doesn't seem to be needed most of the time, but I'm open to hearing if that's not actually the case

Reply:
sorry i forgot we actually push these images separately in the release.yml workflow, so no need to include release here. i guess we can leave this as only main and see how it works out. it's possible we'll end up wanting to not have this filter to ensure building works before merging each feature branch to main. wdyt @sliptype ?

Reply:
Honestly if test-audius-cmd passes on the feature branch we should be golden, so I think we can leave the filter. Let's see if we get into any cases where we merge broken things into main though!